src/plugins/english/mvlempyr.ts

```diff
@@ -0,0 +1,243 @@
+          origin2: this.site.replace(/\/$/, ''),
+          origin: this.site.replace(/\/$/, ''),
+        },
+      }).then(res => res.json())
+    )[0];
+
+    const posts = (
+      await Promise.all(
+        new Array(Math.ceil(tags.count / 500))
+          .fill(0)
+          .map((_, i) => i + 1)
+          .map(page =>
+            fetchApi(
+              this._chapSite +
+                'wp-json/wp/v2/posts?tags=' +
+                tags.id +
+                '&per_page=500&page=' +
+                page,
+              {
+                headers: {
+                  // origin2: this.site.replace(/\/$/, ''),
+                  origin: this.site.replace(/\/$/, ''),
+                },
+              },
+            ).then(res => res.json()),
+          ),
+      )
+    )
+      .flat()
+      .sort((a, b) => a.acf.chapter_number - b.acf.chapter_number);
+
+    return {
+      path: novelPath,
+      name:
+        loadedCheerio('div.image-container.w-embed > img').attr('alt') ||
+        'Untitled',
+      cover: loadedCheerio('div.image-container.w-embed > img').attr('src'),
+      summary: loadedCheerio('div.synopsis.w-richtext').text().trim(),
+      chapters: posts.map(chap => ({
+        name: chap.acf.ch_name,
+        path: 'chapter/' + chap.acf.novel_code + '-' + chap.acf.chapter_number,
+        date: chap.date,
+        chapterNumber: chap.acf.chapter_number,
+      })),
+      status: loadedCheerio('div.novelstatustextmedium').text(),
+      author: loadedCheerio('div.mobileauthorname').text(),
+      genres: loadedCheerio('div.novelgenre.mobile > div > div > a > div')
+        .map((i, el) => loadedCheerio(el).text())
+        .toArray()
+        .join(','),
+    };
+  }
+
+  async parseChapter(chapterPath: string): Promise<string> {
+    const result = await fetchApi(this.site + chapterPath);
+    const body = await result.text();
+
+    const loadedCheerio = parseHTML(body);
+    this.checkCaptcha(loadedCheerio);
+
+    return loadedCheerio('#chapter').html() || '';
+  }
+
+  async searchNovels(
+    searchTerm: string,
+    page: number,
+  ): Promise<Plugin.NovelItem[]> {
+    if (page > 1) return [];
+
+    const allNovels = await this.getAllNovels();
+    const searchResults = allNovels.filter(novel =>
+      novel.description.toLowerCase().includes(searchTerm.toLowerCase()),
+    );
+
+    return searchResults.map(novel => ({
+      name: novel.description,
+      path: 'novel/' + novel.description.replace(/\s/g, '-').toLowerCase(),
+      cover: this._assetsSite + 'images/600/' + novel.name + '.webp',
+    }));
+  }
+
+  async getAllNovels() {
+    if (this._allNovels) {
+      return this._allNovels;
+    }
+    this._allNovels = await this.loadAll();
+    return this._allNovels;
+  }
+
+  async loadAll() {
+    const ret: WPData[] = [];
+    let page = 1;
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const pages = await this.loadNPages(page, 5);
+      page += 5;
```

Comment:
there is around 20 pages so it takes a bit, loading 5 at a time makes it load almost instantly tho

Reply:
(cus the files are relatively small most the time is spent waiting on response not downloading response)